### DISCUSSION TITLE
Model modifications

### DISCUSSION THREAD

**Discussion by Maik7** [joba-1/Corona_Model#3] [General]
English version below.

Ich habe folgende Modifikationen am Modell vorgenommen:

1. Berücksichtigung von Impfverweigerern
2. Berücksichtigung, dass die Impfstoffe für Kinder erst später zugelassen werden
3. 10 Infizierte zu Beginn der Simulation statt 4 (= 1 auf 1000)
4. Es werden bei jedem Lauf immer die selben n Personen infiziert. Dazu wurde die Simulation so geändert, dass diese nie zu den
geimpften zählen. Im Original wurden in späteren Läufen andere Personen infiziert, wenn die ursprünglich vorgesehen geimpft waren. Sicherlich zu vernachlässigen, wenn viele Wiederholungen der Simulation betrachtet werden, kann aber bei einzelnen Läufen relevant sein.

Alle im folgenden gezeigten Verteilungen berücksichtigen, dass für Kinder die Impfstoffe erst später zugelassen werden.
Initiale Zulassung ab 16 Jahre
Zulassung ab 12 Jahren bei 40% Durchimpfung
Zulassung ab 8 Jahren bei 80% Durchimpfung 

Der simulierte Ausbruch beginnt mit 10 infizierten. Also 1 auf 1000, das entspricht der Rate der beim Tübinger Modell erkannten Hintergrundaktivität bei einem Inzidenzwert um 50. Da die Tests nur ~50% der Infizierenden erkennen, könnte das bei einem Inzidenzwert von 25 realistisch sein.

Keine Impfverweigerer bei den durchgezogenen Kurven.
Impfverweigerer in Höhe von 20% bei den gestrichelten Kurven (d.h. die Kurven enden dann bei 70%. 80% und mehr Durchimpfung geht nur durch Überzeugung der Impfverweigerer.

Bezüglich Zahl der Infektionen nähern sich die verschiedenen Strategien deutlich an. Wenn die Simulation halbwegs realistisch ist, ergibt sich, dass wir möglicherweise keine Herdenimmunität durch Impfen erreichen. Ein erheblicher Teil der Impfverweigerer wird sich in einer finalen Welle (nach Lockerung der Maßnahmen, vermutlich ab Herbst/Winter) infizieren.
Ein sehr großer Unterschied ergibt sich aber bei der Zahl der Verstorbenen. 

Wenn die Simulation zutrifft, hat die Regierung genau die richtige Strategie gewählt und sollte zeitnah die Strategie Richtung "jetzt die Aktiven" ändern. Ich gehe davon aus, dass dazu eine Aufhebung der allgemeinen Impfpriorisierung (ggf. noch Schüler /Studenten vorziehen?) ausreicht.
![grafik](https://user-images.githubusercontent.com/83724253/117536810-90736380-affd-11eb-899c-53765baf5664.png)

Hier beispielhaft der Verlauf eines Ausbruchs bei 80% Durchimpfung + durchmachte Infektionen; könnte die Situation  sein, wenn alle die wollen und dürfen geimpft sind. Etwa 40% der nicht-geimpften erkrankt dann noch in dieser Simulation.

![grafik](https://user-images.githubusercontent.com/83724253/117536613-4b9afd00-affc-11eb-99b9-ac70ad80c2c2.png)

Bei diesem Simulationslauf habe ich die Rate der Impfverweigerer moduliert. Verzögerte Zulassung für Kinder wie oben.
Impfverweigerer:
0% durchzogen, 
10% gestrichpunktet
20% gestrichelt
30% gepunktet
![grafik](https://user-images.githubusercontent.com/83724253/117541039-48127080-b012-11eb-87ea-bfa1e5a32263.png)
Letztlich keine Überraschungen in den Verläufen. Es gab bei höheren Impfverweigerern-Quoten bei 60-70% Durchimpfung Läufe, bei denen der Ausbruch erstickte. Grundsätzlich wird das so erwartet, siehe Originalveröffentlichung. Wahrscheinlich ist es rein zufällig so aufgetreten. Evtl. gibt es aber auch eine systematische Ursache. Wenn es mehr Impfverweigerer gibt, werden eher auch Mitglieder anderer Gruppen geimpft. Und nur wenn in allen Gruppen eine hinreichend große Zahl geimpft ist, kann in allen Gruppen ein Ausbruch verhindert werden.  

@joba-1 Ich würde die Erweiterungen gerne hochladen. Habe ich dazu die Rechte bzw. möchtest Du sie mir dazu geben?

I made the following modifications to the model:

1. consideration of vaccination deniers
2. consideration that the vaccines for children will be approved only later
3. 10 infected persons at the beginning of the simulation instead of 4 (= 1 in 1000)
4. the same n persons are always infected in each run. For this purpose, the simulation was changed in such a way that they are never counted among the infected. In the original, other persons were infected in later runs if those originally intended were vaccinated. Certainly negligible when many replicates of the simulation are considered, but may be relevant in individual runs.

All distributions shown below take into account that vaccines are not licensed for children until later.
Initial approval from age 16
Approval from 12 with 40% vaccination coverage
Approval from 8 with 80% vaccination coverage 

The simulated outbreak starts with 10 infected. So 1 in 1000, which is the rate of background activity detected in the Tübingen model at an incidence value around 50. Since the tests only detect 50% of those infected, this could be realistic at an incidence value of 25.

No vaccination deniers at the solid curves.
Vaccine deniers at the 20% level on the dashed curves (i.e., the curves then end at 70%. 80% and more vaccination coverage can only be achieved by convincing the vaccination refusers.

In terms of number of infections, the different strategies converge significantly. If the simulation is halfway realistic, it follows that we may not achieve herd immunity through vaccination. A significant proportion of vaccine refusers will become infected in a final wave (after measures are relaxed, probably starting in the fall/winter).
However, a very large difference arises in the number of deaths. 

If the simulation is true, the government has chosen exactly the right strategy and should promptly change the strategy towards "now the active". I assume that a cancellation of the general vaccination prioritization (if necessary still give priority to pupils /students ?) is sufficient for this.
![grafik](https://user-images.githubusercontent.com/83724253/117536810-90736380-affd-11eb-899c-53765baf5664.png)

Here exemplary the course of an outbreak with 80% vaccination coverage + passed infections; could be the situation if all who want and are allowed are vaccinated. About 40% of the non-vaccinated then still get sick in this simulation.

![grafik](https://user-images.githubusercontent.com/83724253/117536613-4b9afd00-affc-11eb-99b9-ac70ad80c2c2.png)

In this simulation run, I modulated the rate of vaccination deniers. Delayed admission for children as above.
Vaccine deniers:
0% solid, 
10% dash-dotted
20% dashed
30% dotted
![grafik](https://user-images.githubusercontent.com/83724253/117541039-48127080-b012-11eb-87ea-bfa1e5a32263.png)
Ultimately, no surprises in the trajectories. There were runs at higher vaccination refusal rates at 60-70% vaccination coverage where the outbreak was smothered. Basically, this is expected, see original publication. It probably occurred that way purely by chance. However, there may also be a systematic cause. If there are more vaccine deniers, members of other groups are more likely to be vaccinated. And only if a sufficiently large number in all groups are vaccinated, an outbreak can be prevented in all groups.  

Translated with help of www.DeepL.com/Translator (free version)

**Comment by joba-1** [comment]
ich werd das gern aufnehmen. Wär wohl am Besten als eigener Branch? Oder sind das nur neue Notebooks - dann könnten die auch gern in den Hauptbranch

**Reply by Maik7** [reply]
Ist alles im Notebook

**Reply by joba-1** [reply]
das vorhandene würd ich ungern damit „überschreiben“. Benenn es bitte um, dann kannst du einen Pull Request schicken.

**Reply by Maik7** [reply]
Ja, werde ich so machen.

**Reply by Maik7** [reply]
Hm, ich nutze git nur sehr selten. Brauche ich ncoh irgendwelche Rechte, um einen Pull Request zu erstellen? Anscheinend brauche ich einen neuen branch für einen pull request. Ich habe einen branch erstelt und scheitere nun am Hochladen
```
remote: Permission to joba-1/Corona_Model.git denied to Maik7.
fatal: unable to access 'https://github.com/joba-1/Corona_Model/': The requested URL returned error: 403
Pushing to https://github.com/joba-1/Corona_Model
```

**Reply by Maik7** [reply]
Oder vielleicht diekt im Kommentar. Ist ja nur ein File:
[Demo_Vaccination - ModMaik.zip](https://github.com/joba-1/Corona_Model/files/6448261/Demo_Vaccination.-.ModMaik.zip)

**Comment by Maik7** [comment]
Hi,
finally I run all strategies combined with 20% vaccine deniers, 10 initial infected people and age depended approval of the vaccines: 
Initial approval from age 16
Approval from age 12 with 40% vaccination coverage
Approval from age 8 with 80% vaccination coverage

The strategy by age and the strategy old_active (down to age 55s sorted by age, then switch to active) seem to be the best strategies. Maybe its worth to check, if a different age threshold to switch to active has influence in the result; but I assume, that a high number of simulations is needed to get statistically significant results.
![grafik](https://user-images.githubusercontent.com/83724253/117583302-c51f1200-b106-11eb-8d34-4db89f4bd06e.png)

**Reply by joba-1** [reply]
üblicherweise funktioniert das mit Github so: mit deinem Github-User anmelden, dann mein Repo forken. Dein geforktes Repo mit git clone  runterladen und dann lokal ändern. Die Änderungen mit git add … und git commit … lokal übernehmen und mit git push ins eigene, geforkte Repo hochladen. Dann einen neuen Pullrequest erstellen, den ich dann prüfen und ggf. übernehmen kann.
Bei Bedarf könntest du auch noch in deiner lokalen Kopie weitere Änderungen vornehmen, nochmal add, commit und push fügt das dem Pullrequest hinzu.
Puh, ich hoffe ich hab nichts vergessen…
Wenn du das einfach mal ausprobieren willst, helf ich dabei bei Bedarf gern. Die Zip Datei reicht aber auch.

**Reply by Maik7** [reply]
Hi, 
Danke für die ausführliche Antwort. Das probiere ich heute aus.
Gruß
Maik

**Reply by Maik7** [reply]
Danke, pull request ist raus :-)

**Comment by joba-1** [comment]
Schöne Ergebnisse. 
Ich hab nochmal beim RKI Dashboard nachgeschaut: 
über 80: 2200 Tote/100k Einwohner, 
60-79: nur noch 270
35-59: noch 22
Rest: fast keine. 
Da macht eine geringere Altersschranke bzw. Sortierung als ca. 60 Jahre einfach keinen messbaren Sinn mehr - ein früherer Schutz kann da niemanden retten - eine schnellere Eindämmung wäre aber in vielerlei Hinsicht sehr vorteilhaft.

**Reply by Maik7** [reply]
Ich habe nun Deine old-age Strategie mit verschiedenen Altersschranken simuliert (immer in Kombination mit 20% Impfverweigerern und gestaffelter Altersfreigabe der Impfstoffe).
Grundsätzlich zeigt sich genau das erwartete Verhalten, je älter die Altersschranke desto geringer die Zahl der Infektionen. Bei der Zahl der Toten müsste man die Simulation häufiger laufen lassen, so ist doch viel Rauschen zu erkennen. Aber die von dir gewählte Altersschranke von 55 könnte nahe am Optimum sein. 


![grafik](https://user-images.githubusercontent.com/83724253/117864935-a56a2400-b295-11eb-8a6b-1946d9bf90ab.png)
